woocommerce-services.php

```diff
@@ -1205,10 +1205,27 @@ public function add_meta_boxes() {
 			if ( $this->shipping_label->should_show_meta_box() ) {
 				add_meta_box( 'woocommerce-order-label', __( 'Shipping Label', 'woocommerce-services' ), array( $this->shipping_label, 'meta_box' ), null, 'side', 'default' );
 			}
+
+			add_meta_box( 'woocommerce-services-shipping-debug', __( 'Shipping Debug', 'woocommerce-services' ), array( $this, 'shipping_rate_packaging_debug_log_meta_box' ), 'shop_order', 'normal', 'default' );
+		}
+
+		public function shipping_rate_packaging_debug_log_meta_box( $post ) {
+			$order            = wc_get_order( $post );
+			$shipping_methods = $order->get_shipping_methods();
+
+			foreach ( $order->get_shipping_methods() as $method ) {
+				if ( empty( $method['wc_connect_packing_log'] ) ) {
+					continue;
+				}
+
+				echo '<strong>', $method->get_name(), '</strong> <em>(', $method->get_method_id(), ':', $method->get_id(), ')</em>';
```

Comment:
Can we add this information to the packing log, so it's clear to the user what's happening:

- the rate was retrieved using WooCommerce Shipping live rates (so that users have context as to why this packing log is showing up)
- an indication of what `$method->get_name()` means, for example `This is the shipping method used for this rate: $method->get_name()`
- the cost of the rate received (so that users don't have to scroll back up to the packing cost line item)
- the box used for the live rate, so it's easy to find when reading from the top

Reply:
Done in https://github.com/Automattic/woocommerce-services/pull/1542/commits/41e3bdadcaab8e9980e5c99dd205d7f1cfa855c6.

With the exception of:
> the box used for the live rate, so it's easy to find when reading from the top

Can you elaborate on what you're wanting to see here? We could list all of the boxes used in the packing, but that information is already presented in the packing log ("winners").

Reply:
Yeh, I meant all the boxes - the same info as the "Winners". If you think it's too much text, I'm ok to skip that 👍